### ISSUE TITLE
Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by benmccann** [vitest-dev/vitest#317]
### Describe the bug

I'm trying Vitest out for the first time by trying to switch `vite-imagetools` to use it. I ran vitest and got:

>Segmentation fault (core dumped)
error Command failed with exit code 139.

### Reproduction

```
git clone https://github.com/benmccann/imagetools.git
cd imagetools
git checkout vitest
yarn
cd packages/core
yarn test
```

### System Info

command doesn't work because this project uses yarn. I'm hoping to send a PR to switch to pnpm, but that's sort of blocked on this issue

anyway, I'm using:
- Vite 2.7.3
- Vitest 0.0.113

### Used Package Manager

yarn

### Logs

```shell
>Segmentation fault (core dumped)
error Command failed with exit code 139.
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by antfu** [comment]
Not sure if I can debug Segmentation fault, help wanted 😅

**Comment by userquin** [comment]
@benmccann can you test with `0.0.117` version (should be fixed)?

**Comment by Demivan** [comment]
Tried it locally - it still segfaults. It is not an Node error like #319 but an actual segfault.

**Comment by MatthewZito** [comment]
I messed around with this for a bit and can see that the tests are specifically failing when we call `resolveConfig` in generate-configs.spec.ts. 

I don't know much about node internals, but it looks like something about the work being delegated causes the segfault. As far as vitest is concerned, the segfault happens once [createWorkerPool](https://github.com/vitest-dev/vitest/blob/4e3e004501c2b3bbb4e0fa8e18eeca42a380fcfc/packages/vitest/src/node/pool.ts#L89) is invoked. 

I hope this helps :/

**Comment by GalassoLuca** [comment]
I'm having the same issue of the #319 starting with the following error

```
FATAL ERROR: v8::FromJust Maybe value is Nothing.
...
```

It's always happening locally when running Vitest with `--coverage` option, while it's not happening when running it without this flag.

I said locally, because it seems the first time in which I log in my mac it's working, while in circleci it's happening just few times (I think because each time a new vm is created from scratch).

Furthermore
 - when disabling `threads` this issue disappear but the mocks fail to work.
 - when using both `minThreads` and `maxThreads` setted to `1`, the point (suite file logged in the console) that generate the error every time is different.

### Notes
- Vitest: 0.5.9
- Node: 14.9.0
- pnpm: 6.29.1

**Comment by GalassoLuca** [comment]
BTW, one time failed with

```
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the trustlayer-api@1.10.15 test:ci script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/luca/.npm/_logs/2022-03-04T11_38_10_920Z-debug.log
```

And this is the log file
[2022-03-04T11_38_10_920Z-debug.log](https://github.com/vitest-dev/vitest/files/8185732/2022-03-04T11_38_10_920Z-debug.log)

**Comment by oneillsp96** [comment]
Does anyone have a line on this? @benmccann does it seem that Vitest cannot solve these segfault issues until Node addresses the VM issue on their side?

**Comment by staadecker** [comment]
I'm also running into this issue, would be nice to have a fix!

**Comment by quinnsprouseaware** [comment]
I'm running into this issue as well 😞 Hope this gets fixed.

**Comment by sheremet-va** [comment]
We made a change in the latest version, can you check?

**Comment by richardmward** [comment]
>     * when using both `minThreads` and `maxThreads` set to `1`, the point (suite file logged in the console) that generates the error every time is different.

Interestingly the only way I can reliably run the tests is to set `minThreads` / `maxThreads` to `1`. Am on Windows 10, Node 16.14 using Vitest 0.12.3 (although we get it on a Linux based CI box too).

**Comment by GalassoLuca** [comment]
I'm further testing it but with Vitest 0.12.3, v8 FATAL ERROR is not occurring anymore.

This is the project I'm working on:
```
  Test Files  192 passed (192)
  Tests  719 passed | 2 todo (721)
  Time  214.79s (in thread 64.47s, 333.15%)
```

**Comment by benmccann** [comment]
Upgrading to the latest did not help me, but rebasing against `JonasKruckenberg/imagetools` did fix it. I'm not exactly sure what fixed it, but I'm guessing it's because the repo switched from yarn to pnpm

That's good enough for me, so I'll close this issue. If other people are still encountering this I'd recommend filing a new issue with a repository that reproduces your error

**Comment by mikemaccana** [comment]
Hrm using `0.26.2` here and getting Segemntation Faults. I don't want to file a new issue because I don't have a repo to reproduce, but I can say the error happens more often the more test files I run in parallel - this is why I can't reproduce, as running tests individually works. 

`npx vitest run src/backend/*test.ts` - seg fault
`npx vitest run src/backend/[a-u]test.ts` - seg fault most of the time
`npx vitest run src/backend/[a-k]test.ts` - works
`npx vitest run src/backend/some-file.test.ts`  (with any file including the ones at the end of the alphabet) - works

**Comment by loxator** [comment]
bumping as I am facing similar issue as @mikemaccana. If I disable threads, then tests run ok

**Comment by Christopher-Stevers** [comment]
I'm running into the same issue as well.

**Comment by rushil23** [comment]
Facing this issue as well - Re-opening an issue because this one's closed for some reason

**Comment by alice-was-here** [comment]
I'm having this same issue - it occurs in some of our ci/cd tests, but not when I run the same suite of tests locally. 

Originally running version v0.26.2, bumped to v0.30.1 with the same issue.

**Comment by colinparsonsme** [comment]
Dealing with the same issue here too!

**Comment by Max101** [comment]
i have the same issue, anyone tested how to fix this?
